agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/quickpulse/QuickPulseDataCollector.java

```diff
@@ -139,6 +145,7 @@ static CountAndDuration decodeCountAndDuration(long countAndDuration) {
   private final AtomicReference<Counters> counters = new AtomicReference<>(null);
   private final MemoryMXBean memory;
   private final CpuPerformanceCounterCalculator cpuPerformanceCounterCalculator;
+  private volatile QuickPulseHeaderInfo quickPulseHeaderInfo;
```

Comment:
looks like this can just be boolean?

Reply:
My initial thought was to use boolean, but I am thinking this is more cleaner.

Reply:
maybe `QuickPulseStatus` then? (storing header here makes it seem like this class depends on the full header)